core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1043,6 +1045,43 @@ class KafkaApis(val requestChannel: RequestChannel,
     info("Shutdown complete.")
   }
 
+  def handleCreateTopicsRequest(request: RequestChannel.Request) {
+    val createTopicsRequest = request.body.asInstanceOf[CreateTopicsRequest]
+
+    val (validTopics, duplicateTopics) = createTopicsRequest.topics.asScala.partition { case (topic, _) =>
+      !createTopicsRequest.duplicateTopics.contains(topic)
+    }
+
+    def sendResponseCallback(results: Map[String, Errors]): Unit = {
+      val respHeader = new ResponseHeader(request.header.correlationId)
+      if(duplicateTopics.nonEmpty)
+        error("Multiple entries for the same topic were found for the following topics: " + duplicateTopics.keySet.mkString(","))
```

Comment:
This probably should be warn since it's not a server error, but rather a client issue. Also, we probably should give some indication on which client sent this request.